manta-js/package/src/pallets/MantaPayWallet.ts

```diff
@@ -135,6 +135,124 @@ export default class MantaPayWallet
     return result;
   }
 
+  async getAllUtxoList() {
+    const result = await this.baseWallet.wrapWalletIsBusy(async () => {
+      const assetList = this.wasmWallet.asset_list(this.getWasmNetWork());
+      const utxoList: UtxoInfo[] = [];
+      assetList.forEach((item: any) => {
+        utxoList.push({
+          asset: {
+            id: u8aToBn(item.asset.id).toString(),
+            value: item.asset.value.toString(),
+          },
+          identifier: {
+            is_transparent: item.identifier.is_transparent,
+            utxo_commitment_randomness: u8aToBn(
+              item.identifier.utxo_commitment_randomness,
+            ).toString(),
+          },
+        } as UtxoInfo);
+      });
+      return utxoList;
+    });
+    return result;
+  }
+
+  async consolidateTransactionBuild(utxoList: UtxoInfo[]) {
+    const result = await this.baseWallet.wrapWalletIsBusy(
+      async () => {
+        await this.baseWallet.isApiReady();
+        const originUtxoList = utxoList.map((item: UtxoInfo) => {
+          return {
+            asset: {
+              id: bnToU8a(bnToBn(item.asset.id), { bitLength: 256 }),
+              value: nToBigInt(item.asset.value),
+            },
+            identifier: {
+              is_transparent: item.identifier.is_transparent,
+              utxo_commitment_randomness: bnToU8a(
+                bnToBn(item.identifier.utxo_commitment_randomness),
```

Comment:
Shouldn't the encoding here be the same as above or maybe I am missing why format is different

Reply:
Line 151 has this block instead

```rust
            utxo_commitment_randomness: u8aToBn(
              item.identifier.utxo_commitment_randomness,
            ).toString(),
```

Reply:
`getAllUtxoList` is to convert the u8a format into a wallet format that is easier to use

`consolidateTransactionBuild` is the list in `getAllUtxoList` that the user selects, and then revert the format to what wasm expects.